src/qt/nametablemodel.cpp

```diff
@@ -99,8 +105,30 @@ class NameTablePriv
                     continue;
                 }
 
-                const std::string name = maybeName.get_str();
-                const std::string data = maybeData.get_str();
+                // TODO: Properly handle non-ASCII names/data.
+
+                const std::string hexName = maybeName.get_str();
+                std::string name;
+                try
+                {
+                    name = NameTableModel::hexToAscii(QString::fromStdString(hexName)).toStdString();
+                }
+                catch (const InvalidNameString& exc)
+                {
+                    continue;
+                }
+
+                const std::string hexData = maybeData.get_str();
+                std::string data;
+                try
+                {
+                    data = NameTableModel::hexToAscii(QString::fromStdString(hexData)).toStdString();
+                }
+                catch (const InvalidNameString& exc)
+                {
+                    continue;
+                }
+
```

Comment:
Nit:  I think it would be more readable if you merged the two `try-catch` into one, which has the same effect in the end.